### PR TITLE
OCM-11297 | test: automated cases id:38810

### PR DESCRIPTION
**Pull request by msmigiel-rh**
```
ansible-2.9) msmigiel@msmigiel-thinkpadp1gen5:~/Documents/rosa$ ginkgo run -focus 38810 tests/e2e
Running Suite: ROSA CLI e2e tests suite - /home/msmigiel/Documents/rosa/tests/e2e
=================================================================================
Random Seed: 1726839612

Will run 1 of 226 specs
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS•SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS

Ran 1 of 226 Specs in 10.556 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 225 Skipped
PASS

Ginkgo ran 1 suite in 14.278910587s
Test Suite Passed
```

### CODE REVIEW

**Comment by mgahagan73** [comment]
I think we should pass the versions as variables rather than hard coded strings. Over time versions may get added/removed from the list of supported versions so the test may need to be updated in the future. If we use variables it makes the test a little easier to maintain as we only have to change the version values in one place.

**Comment by msmigiel-rh** [comment]
Thank you, I was thinking the same thing after discussing with Ohad. I will look into doing this.

File: tests/e2e/test_rosacli_version.go [hunk 1]
```diff
@@ -104,4 +104,86 @@ var _ = Describe("Get CLI version",
 
 			},
 		)
+
+		It("list versions can work correctly via ROSA cli - [id:38810]",
+			labels.High,
```
Comment (xueli181114) [inline tests/e2e/test_rosacli_version.go:109]: Lack of runtime label.

File: tests/e2e/test_rosacli_version.go [hunk 2]
```diff
@@ -104,4 +104,86 @@ var _ = Describe("Get CLI version",
...
+			func() {
+
+				By("Init the client")
+				rosaClient = rosacli.NewClient()
```
Comment (xueli181114) [inline tests/e2e/test_rosacli_version.go:113]: The client had been init in before each. No need to init again.

File: tests/e2e/test_rosacli_version.go [hunk 3]
```diff
@@ -104,4 +104,86 @@ var _ = Describe("Get CLI version",
...
+				buf, err = rosaClient.Runner.Cmd("list", "version").Run()
+				Expect(err).ToNot(HaveOccurred())
+				stdout = rosaClient.Parser.TextData.Input(buf).Parse().Output()
+
+				By("Check the output of the stable versions")
+				Expect(stdout).To(ContainSubstring("AVAILABLE UPGRADES"))
+				verList, err := versionService.ListAndReflectJsonVersions("stable", false)
+				Expect(err).ToNot(HaveOccurred())
+				for _, v := range verList {
+					Expect(v.ChannelGroup).To(Equal("stable"))
```
Comment (xueli181114) [inline tests/e2e/test_rosacli_version.go:137]: Put `stable` to const .

File: tests/e2e/test_rosacli_version.go [hunk 4]
```diff
@@ -104,4 +104,86 @@ var _ = Describe("Get CLI version",
...
+
+				By("Display the version on the candidate channel")
+				rosaClient.Runner.UnsetArgs()
+				buf, err = rosaClient.Runner.Cmd("list", "version", "--channel-group", "candidate").Run()
+				Expect(err).ToNot(HaveOccurred())
+				stdout = rosaClient.Parser.TextData.Input(buf).Parse().Output()
+
+				By("Check the output of the candidate versions")
+				Expect(stdout).To(ContainSubstring("AVAILABLE UPGRADES"))
+				verList, err = versionService.ListAndReflectJsonVersions("candidate", false)
```
Comment (xueli181114) [inline tests/e2e/test_rosacli_version.go:153]: Put `candidate` to const part.

**Comment by msmigiel-rh** [comment]
Hi @xueli181114 - thanks for the feedback. I've incorporated it all except for `The client had been init in before each. No need to init again.`. I noticed this too and it does not work as intended. There seems to be some bug in the framework as the test case above mine does the same thing, re initializes the rosaClient. Without that, I get the following errors when I run:

```
(ansible-2.9) msmigiel@msmigiel-thinkpadp1gen5:~/Documents/rosa$ ginkgo run -focus 38810 tests/e2e
Running Suite: ROSA CLI e2e tests suite - /home/msmigiel/Documents/rosa/tests/e2e
=================================================================================
Random Seed: 1727088558

Will run 1 of 226 specs
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS
------------------------------
• [FAILED] [0.083 seconds]
Get CLI version [It] list versions can work correctly via ROSA cli - [id:38810] [feature-version, High, ocm-resources]
/home/msmigiel/Documents/rosa/tests/e2e/test_rosacli_version.go:108

Timeline >>
STEP: Init the client @ 09/23/24 06:49:21.361
STEP: Display the version help page @ 09/23/24 06:49:21.361
time=2024-09-23T06:49:21-04:00 level=info msg=Running command: rosa list version -h
time=2024-09-23T06:49:21-04:00 level=info msg=Get Combining Stdout and Stderr is :
List versions of OpenShift that are available for creating clusters.

Usage:
rosa list versions [flags]

Aliases:
versions, version

Examples:
# List all OpenShift versions
rosa list versions

Flags:
--channel-group string   List only versions from the specified channel group (default "stable")
-h, --help                   help for versions
--hosted-cp              Lists only versions that are hosted-cp enabled
-o, --output string          Output format. Allowed formats are [json yaml]

Global Flags:
--color string     Surround certain characters with escape sequences to display them in color on the terminal. Allowed options are [auto never always] (default "auto")
--debug            Enable debug mode.
--profile string   Use a specific AWS profile from your credential file.
--region string    Use a specific AWS region, overriding the AWS_REGION environment variable. (DEPRECATED: Region flag will be removed from this command in future versions)

STEP: Check the output of the help page @ 09/23/24 06:49:21.403
STEP: Display the version on the stable channel @ 09/23/24 06:49:21.403
time=2024-09-23T06:49:21-04:00 level=info msg=Running command: rosa list version
time=2024-09-23T06:49:21-04:00 level=info msg=Get Combining Stdout and Stderr is :
ERR: Failed to create OCM connection: Not logged in, run the 'rosa login' command

[FAILED] in [It] - /home/msmigiel/Documents/rosa/tests/e2e/test_rosacli_version.go:132 @ 09/23/24 06:49:21.444
<< Timeline

[FAILED] Unexpected error:
<*errors.errorString | 0xc0010a8680>:
exit status 1: ERR: Failed to create OCM connection: Not logged in, run the 'rosa login' command

{
s: "exit status 1: ERR: Failed to create OCM connection: Not logged in, run the 'rosa login' command\n",
}
occurred
In [It] at: /home/msmigiel/Documents/rosa/tests/e2e/test_rosacli_version.go:132 @ 09/23/24 06:49:21.444
------------------------------
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS

Summarizing 1 Failure:
[FAIL] Get CLI version [It] list versions can work correctly via ROSA cli - [id:38810] [feature-version, High, ocm-resources]
/home/msmigiel/Documents/rosa/tests/e2e/test_rosacli_version.go:132

Ran 1 of 226 Specs in 0.092 seconds
FAIL! -- 0 Passed | 1 Failed | 0 Pending | 225 Skipped
--- FAIL: TestROSACLIProvider (0.10s)
FAIL

Ginkgo ran 1 suite in 2.794240571s

Test Suite Failed
(ansible-2.9) msmigiel@msmigiel-thinkpadp1gen5:~/Documents/rosa$
```

**Comment by xueli181114** [comment]
@msmigiel-rh The root cause of below error
```
[FAILED] Unexpected error:
<*errors.errorString | 0xc0010a8680>:
exit status 1: ERR: Failed to create OCM connection: Not logged in, run the 'rosa login' command

{
s: "exit status 1: ERR: Failed to create OCM connection: Not logged in, run the 'rosa login' command\n",
}
occurred
```
is caused by your local rosa didn't login. It's not part of frame work. But a [prerequisite](https://github.com/openshift/rosa/tree/master/tests#prerequisite)  for the automation running .

**Comment by xueli181114** [comment]
I know what's the problem. Remove below code from beforeEach, I didn't see why it is put in BeforeEach. I believe specified case need it since the case will write specified info to the configFile. But It's not required for all cases. Let's make it happen in specified cases those need it. 
```
configFile, err := common.CreateTempOCMConfig()
			Expect(err).ToNot(HaveOccurred())
			rosaClient.Runner.AddEnvVar("OCM_CONFIG", configFile)
```

**Comment by msmigiel-rh** [comment]
Thank you @xueli181114 , that is working. I have made the changes.

**Pull request closed without merging**